phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java

```diff
@@ -1519,6 +1536,17 @@ private PTable getTableFromCells(List<Cell> tableCellList, List<List<Cell>> allC
         builder.setParentSchemaName(parentSchemaName != null ? parentSchemaName :
             oldTable != null ? oldTable.getParentSchemaName() : null);
 
+        transformRecord = Transform.getTransformRecord(env.getConfiguration(), tableType, schemaName, tableName, dataTableName, tenantId, parentLogicalName);
```

Comment:
Wonder if it would be better to have an IS_TRANSFORMING flag or "TRANSFORM_NEW_TABLE" column on the table row in syscat to avoid the extra lookup. That would add extra complexity to the transform start and completion logic to populate and remove the syscat value, of course.

Reply:
It's not just hotspotting perf problems either -- this adds SYSTEM.TRANSFORM to the list of tables (meta and SYSTEM.CATALOG) that can kill the cluster if they go down.

Reply:
@gjacoby126  Yes, you are right. I actually changed this with my TransformMonitor PR which I didn't raise yet. Let me get those changes as part of this PR. In that change, instead of a column, I use a Link type. The old table has a link to new table.